btree/btree.c

```diff
@@ -4306,8 +4204,20 @@ int64_t btree_open_tree_tick(struct m0_sm_op *smop)
 		return tree_get(&oi->i_nop, &oi->i_nop.no_addr, P_ACT);
 
 	case P_ACT:
+		if (!oi->i_nop.no_tree->t_type)
+			oi->i_nop.no_tree->t_type = bop->b_data.bt;
+
+		/**
+		 * When tree_open() is called after tree_close() and the tree
+		 * descriptor associated with the tree is freed during the
+		 * tree_close(), we will get a new tree_descriptor which wont
+		 * contain the tree_height. This data is then filled using the
+		 * tree's t_height.
+		 */
+		if (oi->i_nop.no_tree->t_height == 0)
```

Comment:
It will be better to update height as per the height stored in be node. The `if` check can be removed.

Reply:
Ok

Reply:
I think we will need to keep this as after freeing the tree_descriptor and again opening the tree, this is the only way to get the  tree height as we dont store height in BE node.

Reply:
`bop->b_data.tree->t_height`  is a user provided parameter. Suppose you have a tree and its height increased to 5. Then you called tree_close and shutdown the cluster. Now you boot up the cluster and call tree_open. This time you do not have any idea about tree height. You have to read the m0_btree:t_height(which is in be segment) and update the td:t_height and use it.